main.go

```diff
@@ -1687,48 +1690,137 @@ func recordedAsNotObfuscated(obj types.Object) bool {
-		node, ok := n.(*ast.Ident)
-		if !ok {
-			return true
-		}
+func (tf *transformer) resolveImportSpec(imp *ast.ImportSpec) *types.Package {
+	obj, ok := tf.info.Implicits[imp]
+	if !ok {
+		obj = tf.info.Defs[imp.Name] // renamed or dot import
+	}
+	if obj == nil {
+		return nil
+	}
 
-		uses, ok := tf.info.Uses[node]
-		if !ok {
-			return true
-		}
+	pkgObj, ok := obj.(*types.PkgName)
+	if !ok {
+		panic(fmt.Sprintf("unexpected object type for %s: %v", imp.Path.Value, obj))
+	}
+	return pkgObj.Imported()
+}
 
-		if pkg := uses.Pkg(); pkg != nil {
-			usedImports[pkg.Path()] = true
+// isSafeForInstanceType returns true if the passed type is safe for var declaration
+// Unsafe types: generic structs, funcs or interfaces, type constraint
+func isSafeForInstanceType(typ types.Type) bool {
+	switch t := typ.(type) {
+	case *types.Named:
+		if t.TypeParams().Len() > 0 {
+			return false
 		}
+		return isSafeForInstanceType(t.Underlying())
+	case *types.Signature:
+		return t.TypeParams().Len() == 0
+	case *types.Interface:
+		return t.IsMethodSet()
+	}
+	return true
+}
 
-		return true
-	})
-
+func (tf *transformer) makeImportsUsed(file *ast.File) {
 	for _, imp := range file.Imports {
 		if imp.Name != nil && imp.Name.Name == "_" {
 			continue
 		}
 
-		path, err := strconv.Unquote(imp.Path.Value)
-		if err != nil {
-			panic(err)
+		pkg := tf.resolveImportSpec(imp)
+		if pkg == nil {
+			panic(fmt.Sprintf("import %s not found", imp.Path.Value))
 		}
 
-		// The import path can't be used directly here, because the actual
-		// path resolved via go/types might be different from the naive path.
-		lpkg, err := listPackage(path)
-		if err != nil {
-			panic(err)
-		}
+		generated := false
+		scope := pkg.Scope()
+		for _, name := range scope.Names() {
+			if !token.IsExported(name) {
+				continue
+			}
+			obj := scope.Lookup(name)
+			if obj == nil {
+				panic(fmt.Sprintf("%s not found in %s", name, imp.Path.Value))
+			}
+			if !isSafeForInstanceType(obj.Type()) {
+				continue
+			}
 
-		if usedImports[lpkg.ImportPath] {
-			continue
+			nameIdent := ast.NewIdent(name)
+			getFullName := func() ast.Expr {
+				if imp.Name == nil {
+					return &ast.SelectorExpr{
+						X:   ast.NewIdent(pkg.Name()),
+						Sel: nameIdent,
+					}
+				}
+				if imp.Name.Name == "." {
+					return nameIdent
+				}
+				return &ast.SelectorExpr{
+					X:   ast.NewIdent(imp.Name.Name),
+					Sel: nameIdent,
+				}
+			}
+
+			var decl *ast.GenDecl
+			switch t := obj.(type) {
+			case *types.Const:
+				valueExpr := getFullName()
+				if val := t.Val(); val.Kind() == constant.Int {
+					// On 32-bit architectures, reference to untyped int constant with a value greater than MaxInt32 gives overflow error
+					castName := "uint64"
+					if constant.Compare(val, token.LSS, constant.MakeInt64(0)) {
+						castName = "int64"
+					}
+					valueExpr = ah.CallExpr(ast.NewIdent(castName), valueExpr)
+				}
+
+				// var _ = <value> or uint64(<value>)
```

Comment:
I wonder if using `var _ interface{} = value` removes the need for the int64/uint64 casts.

Another option would be to use `const _ = value` for constants, so that you don't have to worry about overflows. That's probably easier.

Reply:
First option does not help. But variant with `const _` is good